indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -317,6 +322,157 @@ public void handle()
     }
   }
 
+  // same as submit supervisor logic
+  private class DynamicAllocationTasksNotice implements Notice
+  {
+    @Override
+    public void handle()
+    {
+      lock.lock();
+      try {
+        long nowTime = System.currentTimeMillis();
+        long minTriggerDynamicFrequency = Long.parseLong(String.valueOf(dynamicAllocationTasksProperties.getOrDefault("minTriggerDynamicFrequencyMillis", 1200000)));
+        // Only queue is full and over minTriggerDynamicFrequency can trigger scale out/in
+        // max(minTriggerDynamicFrequency, metricsCollectionRangeMillis)
+        if (spec.isSuspended()) {
+          log.info("[%s] supervisor is suspended, skip to check dynamic allocate task logic", dataSource);
+          return;
+        }
+        log.info("PendingCompletionTaskGroups is : " + pendingCompletionTaskGroups);
+        for (CopyOnWriteArrayList list : pendingCompletionTaskGroups.values()) {
+          if (!list.isEmpty()) {
+            log.info("Still hand off tasks unfinished, skip to do scale action [" + pendingCompletionTaskGroups + "]");
+            return;
+          }
+        }
+        if (nowTime - dynamicTriggerLastRunTime < minTriggerDynamicFrequency) {
+          log.info("NowTime - dynamicTriggerLastRunTime is [" + (nowTime - dynamicTriggerLastRunTime) + "]. Defined minTriggerDynamicFrequency is [" + minTriggerDynamicFrequency + "] , CLAM DOWN NOW !");
+          return;
+        }
+        if (!queue.isAtFullCapacity()) {
```

Comment:
does this mean we have not collected enough historical lag data to decide on scale in/scale out? I think the log can be updated to be more descriptive since it may not be obvious to log reader why it matters that queue is not full

Reply:
Done.